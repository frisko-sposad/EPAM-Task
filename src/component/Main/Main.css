.main {
  margin: 0 auto;
  display: flex;
  flex-wrap: wrap;  
  padding: 1rem 3rem;
}
