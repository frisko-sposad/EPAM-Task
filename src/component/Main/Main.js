import React from 'react';
import style from './Main.module.css';
import MovieItem from './MovieItem/MovieItem';

const Main = () => {
  return (
    <main className={style.Main}>
      <div className={style.main__container}>
        <MovieItem />
        <MovieItem />
        <MovieItem />
        <MovieItem />
        <MovieItem />
        <MovieItem />
        <MovieItem />
        <MovieItem />
      </div>
    </main>
  );
};

export default Main;
