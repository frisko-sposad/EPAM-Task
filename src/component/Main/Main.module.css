.Main {
  display: flex;  
  flex-wrap: wrap;
  justify-content: center;
  padding: 1rem;  
}

.main__container {
  display: flex;
  flex-wrap: wrap;
  max-width: 1060px;  
}
