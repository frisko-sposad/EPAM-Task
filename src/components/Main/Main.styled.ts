import styled from 'styled-components';

export const MainContainer = styled.main`
  margin: 0 auto;
  display: flex;
  flex-wrap: wrap;
  padding: 1rem 3rem;
  justify-content: center;
`;

export const Container = styled.div`
  padding: 0 50px;
  display: flex;
  justify-content: center;
  flex-direction: column;
`;
