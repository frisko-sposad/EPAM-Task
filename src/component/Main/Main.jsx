import React from 'react';
import './Main.css';
import MovieItem from './MovieItem/MovieItem';

const Main = ({ movieBase }) => (
  <main className="main">
    {movieBase.map(({ id, name, year, genre, image }) => (
      <MovieItem key={id} name={name} year={year} genre={genre} image={image} />
    ))}
  </main>
);

export default Main;
