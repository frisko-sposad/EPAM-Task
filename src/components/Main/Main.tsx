import React, { memo } from 'react';
import './Main.css';
import MovieItem, { MovieItemProps } from './MovieItem/MovieItem';

<<<<<<< HEAD
<<<<<<< HEAD
interface MainProps {
  movieBase: (MovieItemProps & { id: number })[];
}

const Main = ({ movieBase }: MainProps) => (
  <main className="main">
    {movieBase.map(({ id, name, year, genre, image }) => (
=======
const Main = ({ movieBase }: { movieBase:MovieItemProps[] }) => (
  <main className="main">    
=======
const Main = ({ movieBase }: { movieBase: MovieItemProps[] }) => (
  <main className="main">
>>>>>>> 31bd3fc... lint jsx -> tsx,
    {movieBase.map(({ id, name, year, genre, image }: MovieItemProps) => (
>>>>>>> 7b7863b... fixed bugs,
      <MovieItem key={id} name={name} year={year} genre={genre} image={image} />
    ))}
  </main>
);

export default memo(Main);
