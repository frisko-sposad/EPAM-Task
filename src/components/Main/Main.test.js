import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';
import movieBase from '../MovieBase';

describe('Render Main.', () => {
  it('Main', () => {
    const component = shallow(<Main movieBase={movieBase} />);
    expect(component).toMatchSnapshot();
  });
});
