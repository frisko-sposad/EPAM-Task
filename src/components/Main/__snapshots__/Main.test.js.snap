// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main. Main 1`] = `
<main
  className="main"
>
  <MovieItem
    genre="fantasy"
    image="https://avatars.mds.yandex.net/get-kinopoisk-image/1704946/57375a05-4883-4994-b33b-8d60741a0706/300x450"
    key="1"
    name="The Witcher"
    year="2019"
  />
  <MovieItem
    genre="Drama"
    image="https://d2ajy4iry6zk4j.cloudfront.net/pubvat/81277950/thumbnail/medium/pkdhjf35g"
    key="2"
    name="Elona Holmes"
    year="2019"
  />
  <MovieItem
    genre="Drama"
    image="https://avatars.mds.yandex.net/get-kinopoisk-image/1898899/4c78b2e3-ffed-4985-bbc2-642952c92f5e/300x450"
    key="3"
    name="1+1"
    year="2011"
  />
  <MovieItem
    genre="Drama"
    image="https://avatars.mds.yandex.net/get-kinopoisk-image/1946459/84934543-5991-4c93-97eb-beb6186a3ad7/300x450"
    key="4"
    name="Joker"
    year="2019"
  />
  <MovieItem
    genre="thriller"
    image="https://avatars.mds.yandex.net/get-kinopoisk-image/1946459/910a3261-ff47-4cc7-866c-6e0388ef3ecc/300x450"
    key="5"
    name="Blade Runner"
    year="2017"
  />
  <MovieItem
    genre="Drama"
    image="https://d2ajy4iry6zk4j.cloudfront.net/pubvat/81277950/thumbnail/medium/pkdhjf35g"
    key="6"
    name="Elona Holmes"
    year="2019"
  />
  <MovieItem
    genre="fantasy"
    image="https://avatars.mds.yandex.net/get-kinopoisk-image/1704946/57375a05-4883-4994-b33b-8d60741a0706/300x450"
    key="7"
    name="The Witcher"
    year="2019"
  />
  <MovieItem
    genre="Drama"
    image="https://d2ajy4iry6zk4j.cloudfront.net/pubvat/81277950/thumbnail/medium/pkdhjf35g"
    key="8"
    name="Elona Holmes"
    year="2019"
  />
  <MovieItem
    genre="fantasy"
    image="https://avatars.mds.yandex.net/get-kinopoisk-image/1704946/57375a05-4883-4994-b33b-8d60741a0706/300x450"
    key="9"
    name="The Witcher"
    year="2019"
  />
  <MovieItem
    genre="Drama"
    image="https://d2ajy4iry6zk4j.cloudfront.net/pubvat/81277950/thumbnail/medium/pkdhjf35g"
    key="10"
    name="Elona Holmes"
    year="2019"
  />
</main>
`;
